js/map_utils.js

```diff
@@ -8,18 +8,33 @@ var zoom_scale;
 function addStates(map, stateBounds) {
 
   // add states
-  map.select('#state-bounds')
+  map.select('#state-bounds-defs')
     .selectAll( 'path' )
     .data(stateBounds.features, function(d) {
       return d.properties.STATE_ABBV;
     })
     .enter()
     .append('path')
+    .attr('id', function(d) {
+      return d.properties.STATE_ABBV+'-pattern';
+    })
+    .attr('d', buildPath);
+    
+  // add states
+  map.select('#state-bounds')
+    .selectAll( 'use' )
+    .data(stateBounds.features, function(d) {
+      return d.properties.STATE_ABBV;
+    })
+    .enter()
+    .append('use')
     .classed('state', true)
     .attr('id', function(d) {
       return d.properties.STATE_ABBV;
     })
-    .attr('d', buildPath);
+    .attr('xlink:href', function(d) {
+      return '#'+d.properties.STATE_ABBV+'-pattern';
+    });
```

Comment:
note that state bounds are not entirely static once they're loaded - they switch between low-resolution (for national view) and higher resolution (for state view). The `addStates` function defines the initialization of these boundaries to the low-resolution dataset, while the `updateStateBounds` function (called in `updateStates`, which is called in `updateView`, anytime you zoom anywhere) drives the switches between these two datasets: https://github.com/USGS-VIZLAB/water-use-15/blob/master/js/states.js#L24-L29.
* To keep the first figure functioning well, the obvious solution to me is to modify the `xlink:href` instead of `data` and `d` within `updateStateBounds` (https://github.com/USGS-VIZLAB/water-use-15/blob/master/js/states.js#L24-L29).
* This one change would make it so that figure 4 will sometimes show low-res and sometimes show high-res state boundaries, depending on the current zoom state of figure 1.
* If we'd like figure 4 to always show low-res state boundaries, I think we need two sets of polygon definitions in `state-bounds-defs` (or a new defs collection, `state-bounds-defs-hi-res`), and then updateStateBounds should switch which definitions are pointed to rather than building a new path. Making this change might actually give us a performance boost to figure 1 as a nice side effect, because we won't be reprojecting each time we switch between national and state views, though it's possible that switching which element is `use`d is more expensive than I think - I say let's just try it and try to notice whether it's a net speedup or slowdown for fig 1.
* If we do create two collections or subcollections of polygon definitions, let's continue to pay attention to when each of those collections gets created. The hi-res state bounds are neither downloaded nor added into the DOM until the first time you zoom in, which I think is a good thing for initial page load time, so I'd like to retain that. I think this means that the building of `state-bounds-defs-hi-res` (or whatever it's naemd) should occur within `updateStateData` right after the data are loaded, probably right here: https://github.com/USGS-VIZLAB/water-use-15/blob/d4fcf0182efbbf34cb3aa173d96d599f5c49095c/js/states.js#L13.

Reply:
Good stuff! I'm thankful for peer review, because I totally whiffed on that one. Changes coming. 

I think I will try putting both resolutions in `use` first and see how that goes. I will probably wait on any class changes because I think there will be a bunch more changes that could wipe that out - specifically, my solution to rely on hovering over the bar doesn't work on mobile.